plugins/module_utils/CertACLHelper.cs

```diff
@@ -0,0 +1,262 @@
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using System.Security.AccessControl;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+using System.Security.Principal;
+
+//TypeAccelerator -Name Ansible.Windows.CertAclHelper.CryptHandle -TypeName CryptHandle
+//TypeAccelerator -Name Ansible.Windows.CertAclHelper.SafeSecurityDescriptorPtr -TypeName SafeSecurityDescriptorPtr
+//TypeAccelerator -Name Ansible.Windows.CertAclHelper.CertAccessRights -TypeName CertAccessRights
+//TypeAccelerator -Name Ansible.Windows.CertAclHelper.CertAclHelper -TypeName CertAclHelper
+
+namespace ansible_collections.ansible.windows.plugins.module_utils.CertACLHelper
+{
+    class CryptHandle : SafeHandleZeroOrMinusOneIsInvalid
+    {
+        public CryptHandle()
+            : base(true)
+        {
+        }
+
+        public CryptHandle(IntPtr handle)
+            : base(true)
+        {
+            this.SetHandle(handle);
+        }
+
+        [DllImport("advapi32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool CryptReleaseContext(IntPtr safeProvHandle, uint dwFlags);
+
+        protected override bool ReleaseHandle()
+        {
+            return CryptReleaseContext(this.handle, 0);
```

Comment:
The docs indicate that if this is a CNG key then `NCryptFreeObject` should be used instead.

Reply:
you're absolutely correct, I added that branch